File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode node = head;
        ListNode temp = null;
        ListNode tempNew = null;
        while(node != null){
            temp = node.next;
            node.next = tempNew;            
            tempNew = node;
            node = temp;
        }
        return tempNew;
    }
}

File: Data Structures & Algorithms/climbing-stairs/submission-0.cs
public class Solution {
    public int ClimbStairs(int n) {     
        return function(n, 0);
    }
    private int function(int n, int i){
        if (i >= n) return i == n ? 1 : 0;
        return function(n, i + 1) + function(n, i+2);
    }
}
